include/openthread/srp_server.h

```diff
@@ -84,11 +84,15 @@ typedef enum
  *
  * Address mode specifies how the address and port number are determined by the SRP server and how this info is
  * published in the Thread Network Data.
+ *
+ * `OT_SRP_SERVER_ADDRESS_MODE_UNICAST_SKIP_PUBLISH` option is intended for testing and specific demo pruposes.
+ * Using this option will be non-compliant with the Thread specification.
  */
 typedef enum otSrpServerAddressMode
 {
-    OT_SRP_SERVER_ADDRESS_MODE_UNICAST = 0, ///< Unicast address mode.
-    OT_SRP_SERVER_ADDRESS_MODE_ANYCAST = 1, ///< Anycast address mode.
+    OT_SRP_SERVER_ADDRESS_MODE_UNICAST              = 0, ///< Unicast address mode. Use Network Data publisher.
+    OT_SRP_SERVER_ADDRESS_MODE_ANYCAST              = 1, ///< Anycast address mode. Use Network Data publisher
+    OT_SRP_SERVER_ADDRESS_MODE_UNICAST_SKIP_PUBLISH = 2, ///< Unicast address mode. Immediately add to Network Data.
```

Comment:
nit: Probably "SKIP_PUBLISH" means skipping the publisher, but this makes think it will skip publishing the service entry... What about rename it to "FAST_PUBLISH" or "IMMEDIATE_PUBLISH"?

Reply:
Good point. Thanks.

I changed it to `ADDRESS_MODE_UNICAST_FORCE_ADD`. It is shorter, and I want to name it such that it is clearer that it overrides the normal behavior.